release/models/network-instance/openconfig-route-summary.yang

```diff
@@ -0,0 +1,108 @@
+module openconfig-route-summary {
+  yang-version "1";
+
+  namespace "http://openconfig.net/yang/route-summary";
+
+  prefix "oc-rtsummary";
+
+  import openconfig-extensions { prefix "oc-ext"; }
+  import openconfig-yang-types { prefix "oc-yang-types"; }
+  import openconfig-policy-types { prefix "oc-pol-types"; }
+  import openconfig-network-instance { prefix "oc-ni"; }
+
+  organization
+    "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module provides summary of route counts per route type for each network
+    instance.";
+
+  oc-ext:openconfig-version "1.0.0";
+
+  revision "2022-12-21" {
+    description
+      "Initial version.";
+    reference "1.0.0";
+  }
+
+  grouping route-counter {
+    description
+      "Route type counter item.";
+
+    leaf source {
+      description
+        "Route type that keys the route count list.";
+
+      type identityref {
+        base "oc-pol-types:INSTALL_PROTOCOL_TYPE";
+      }
+    }
+
+    leaf total-count {
+      description
+        "Total number of routes in the AFT for the given source. This count includes
+        routes which may not be installed due to hardware constraints.";
+      type oc-yang-types:counter64;
+    }
+
+    leaf installed-count {
+      description
+        "Number of AFT routes installed in the hardware for the given source.";
+      type oc-yang-types:counter64;
+    }
+  }
+
+  grouping route-summary {
+    description
+      "A summary of route count by route type.";
+
+    list route-count {
+      key "source";
+
+      description
+        "Route types that keys the route count list.";
+
+      leaf source {
+        type leafref {
+          path "../state/source";
+        }
+        description
+          "Reference to the source of he route.";
+      }
+
+      container state {
+        description
+          "State parameters for the route summary list entry.";
+        uses route-counter;
+      }
+    }
+  }
+
+  augment "/oc-ni:network-instances/oc-ni:network-instance/oc-ni:afts/oc-ni:ipv6-unicast" {
+    description
+      "Augment the network-instance model with the route summary container for
+      IPv4.";
+
+    container summary {
+      uses route-summary;
+      description
+        "IPv4 route summary for the network instance.";
+    }
+  }
+
+  augment "/oc-ni:network-instances/oc-ni:network-instance/oc-ni:afts/oc-ni:ipv4-unicast" {
+    description
+      "Augment the network-instance model with the route summary container for
+      IPv6.";
+
+    container summary {
```

Comment:
+1 to move to tables/table

Reply:
Per the above, I think with this clarification it doesn't need to be under `tables/table`.

Reply:
Sorry, thinking-through-fingers here -- it could be under `tables/table` because of it being keyed by source, which is neater than having a new "protocol" specific thing under AFT (which doesn't really key by protocol today).

I think some discussion as to the pros and cons of each is probably needed. It seems like the counters you might want are:

* RIB count (total number of things in this protocol's table)
* Best path count (number of those things that are actually bestpath and hence in the AFT)
* Installed count (how many actually made it to hardware)

if all these are included putting it in `tables/table` makes more sense I think, but am very open to discussion.

Reply:
@robshakir In a scenario where 100 routes ([1.1.1.1/32](http://1.1.1.1/32) .... [1.1.1.100/32](http://1.1.1.100/32)) in IPv4 FIB  & following is the count per protocol:
BGP - 50
OSPF - 30
ISIS - 20:

Following is the output for both proposals
proposal 1:
========
```
table/
   table[]
       BGP ->50
       OSPF -> 30
       ISIS -> 20
afts/
   ipv4-unicast/
      ipv4-entry/
         1.1.1.1/32
              .
              .
          1.1.1.100/32
```
proposal 2:
=========
```
afts/
   ipv4-unicast/
      ipv4-entry/
         1.1.1.1/32
              .
              .
          1.1.1.100/32
      summary
          BGP ->50
          OSPF -> 30
          ISIS -> 20
```
Proposal 2 is more organized in our opinion since all the information about IPv4 FIB can be find under the same container. Also, Proposal 1 is a little unintuitive since someone looking at this output really needs to understand which state's summary it is representing.

Reply:
The problem here is that you’re just looking at one count here. Consider two — `rib-count` and `installed-count` with the definitions as described above.

It makes little sense to me to have a count for BGP under AFT of `rib-count`, because we define the AFT as the installed routes. It makes sense to have `installed-count` and `bestpath-count` potentially though. If we have `rib-count` somewhere else in the model, then is this unintuitive that these are in different places?

I’d welcome others’ opinions here.